### PR TITLE
chore: plugin-server-mode use stronger typing

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

It's quite annoying how we're using strings in different places when making any changes

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: plugin-server/src/types.ts [hunk 1]
```diff
@@ -72,6 +72,34 @@ export enum KafkaSaslMechanism {
     ScramSha512 = 'scram-sha-512',
 }
 
+export enum PluginServerMode {
+    ingestion = 'ingestion',
+    ingestion_overflow = 'ingestion-overflow',
```
Comment (hazzadous) [inline plugin-server/src/types.ts:77]: To be consistent with variable casing we could use camel case instead of snake case here.

**Review by hazzadous — APPROVED** [review]
Would this have caught the issue btw?

An aside: I would consider just dropping the PLUGIN_SERVER_MODE abstraction and rather we can explicitly list out the things we want to start instead of having this mapping. This way I think we could more easily roll things out in a way that doesn't require coordination between what the chart thinks defines a mode and what the app things it is. But that's a bigger change than be concerned about here.

File: plugin-server/src/config/config.ts [hunk 2]
```diff
@@ -168,25 +175,6 @@ export function overrideWithEnv(
     }
     const newConfig: PluginsServerConfig = { ...tmpConfig }
 
-    if (
-        ![
-            'ingestion',
-            'async',
-            'async-onevent',
-            'async-webhooks',
```
Comment (tiina303) [inline plugin-server/src/config/config.ts:176]: yup, I wouldn't have needed to manually add them here

**Comment by tiina303** [comment]
> An aside: I would consider just dropping the PLUGIN_SERVER_MODE abstraction and rather we can explicitly list out the things we want to start instead of having this mapping. This way I think we could more easily roll things out in a way that doesn't require coordination between what the chart thinks defines a mode and what the app things it is. But that's a bigger change than be concerned about here.

good idea - should this be a list of capabilities then or each one separately as the input

**Comment by tiina303** [comment]
I'll merge it in as is and then work on improving upon it later as that already makes it nicer to add the ingestion backfill consumer we need fast for the backfill coming up

**Pull request merged**